services/weatherApi.service.go

```diff
@@ -0,0 +1,174 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gofiber/fiber/v2"
+	"gorm.io/gorm"
+	"net/http"
+	"project_weather/controllers"
+	"project_weather/generated/dao/model"
+	"project_weather/repositories"
+	"project_weather/resources/dto"
+	"time"
+)
+
+const (
+	forecastUrl         = "http://api.weatherapi.com/v1/forecast.json?key=%s&q=%s&days=%v&aqi=no&alerts=no"
+	currentTimeUrl      = "http://api.weatherapi.com/v1/current.json?key=%s&q=%s&aqi=no"
+	templateDate        = "2006-01-02"
+	templateDateAndTime = "2006-01-02 15:04"
+)
+
+type WeatherApiService interface {
+	GetRoutes() []controllers.Route
+	GetWeatherByCity(ctx *fiber.Ctx) error
+	fetchWeatherData(url string) (*dto.WeatherDTO, error)
```

Comment:
Normally the interface includes the methods that you expect to be called outside of the class? If you don't plan to call `fetchWeatherData` and `formatWeatherData` from 'outside' better remove them